File: bet-event-aggregator/src/main/java/com/piter/bet/event/aggregator/domain/Bet.java
package com.piter.bet.event.aggregator.domain;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.extern.jackson.Jacksonized;

@Jacksonized
@Builder
public record Bet(
    @NotNull
    String id,
    @NotNull
    @Valid
    MatchResult matchPredictedResult,
    @NotNull
    @Valid
    Match match,
    @NotNull
    @Valid
    User user,
    @NotNull
    @Valid
    BetResult betResult) {

}

File: bet-frontend/src/app/model/bet.ts
import {Match, MatchResult} from "./match";
import {User} from "./user";

export class Bet {
  id: string;
  matchPredictedResult: MatchResult;
  match: Match;
  user: User;
  betResult: BetResult;

  constructor(id: string,
              matchPredictedResult: MatchResult,
              match: Match,
              user: User,
              betResult: BetResult) {
    this.id = id;
    this.matchPredictedResult = matchPredictedResult;
    this.match = match;
    this.user = user;
    this.betResult = betResult;
  }
}

export class BetResult {
  status: string;
  points: number;

  constructor(status: string, points: number) {
    this.status = status;
    this.points = points;
  }
}
